cmd/agent.go

```diff
@@ -699,13 +702,100 @@ Examples:
 	},
 }
 
+var agentTestCmd = &cobra.Command{
+	Use:   "test",
+	Short: "Test an agent",
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := env.NewLogger(cmd)
+
+		ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, syscall.SIGINT, syscall.SIGTERM)
+		defer cancel()
+		theproject := project.EnsureProject(ctx, cmd)
+
+		agentID, _ := cmd.Flags().GetString("agent-id")
+		payload, _ := cmd.Flags().GetString("payload")
+		local, _ := cmd.Flags().GetBool("local")
+
+		if agentID == "" {
+			keys, state := reconcileAgentList(logger, cmd, theproject.APIURL, theproject.Token, theproject)
+			if len(keys) == 0 {
+				tui.ShowWarning("no Agents found")
+				tui.ShowBanner("Create a new Agent", tui.Text("Use the ")+tui.Command("agent new")+tui.Text(" command to create a new Agent"), false)
+				return
+			}
+			var options []tui.Option
+			for _, v := range keys {
+				options = append(options, tui.Option{
+					ID:   state[v].Agent.ID,
+					Text: tui.PadRight(state[v].Agent.Name, 20, " ") + tui.Muted(state[v].Agent.ID),
+				})
+			}
+			agentID = tui.Select(logger, "Select an agent", "Select the agent you want to test", options)
+			agentID = strings.Replace(agentID, "agent_", "", 1)
+		}
+
+		if payload == "" {
+			payload = tui.Input(logger, "Enter the payload to send to the agent", "{\"hello\": \"world\"}")
+		}
+
+		apikey, err := agent.GetApiKey(context.Background(), logger, theproject.APIURL, theproject.Token, agentID)
+		if err != nil {
+			errsystem.New(errsystem.ErrApiRequest, err, errsystem.WithContextMessage("Failed to get agent API key")).ShowErrorAndExit()
+		}
+		endpoint := fmt.Sprintf("%s/webhook/%s", theproject.TransportURL, agentID)
+		if local {
+			port, _ := dev.FindAvailablePort(theproject)
+			endpoint = fmt.Sprintf("http://localhost:%d/agent_%s", port, agentID)
+		}
+
+		// use http package to send a POST request to the agent
+		req, err := http.NewRequest("POST", endpoint, strings.NewReader(payload))
+		if err != nil {
+			logger.Fatal("Failed to create request: %s", err)
+		}
+		// check if payload is json
+		if json.Valid([]byte(payload)) {
+			req.Header.Set("Content-Type", "application/json")
+		} else {
+			req.Header.Set("Content-Type", "application/octet-stream")
```

Comment:
we might want to allow the user to pass in the type via a flag. since most of the time i want to use either JSON or Text